private/bufpkg/bufmodule/bufmoduleref/bufmoduleref.go

```diff
@@ -364,6 +371,46 @@ func ValidateModulePinsUniqueByIdentity(modulePins []ModulePin) error {
 	return nil
 }
 
+// ValidateModulePinsConsistentDigests verifies that module pins to the same commit don't change digests.
+// This is important to avoid MITM issues, where the module digest stored in a buf.lock file doesn't match
+// the module pin returned from the BSR.
+func ValidateModulePinsConsistentDigests(
+	ctx context.Context,
+	bucket storage.ReadWriteBucket,
+	modulePins []ModulePin,
+) error {
+	currentConfig, err := buflock.ReadConfig(ctx, bucket)
```

Comment:
This is why we've changed the interface to return a `ReadWriteBucket` instead of a `WriteBucket`. Before updating the `buf.lock` file, we want to verify the current contents vs. the new contents.

Reply:
Maybe take a `bucket storage.ReadBucket` instead? Write isn't needed in this particular function.